### PR TITLE
fix(SearchField): accessibility - do not exclude clear button from tab order

### PR DESCRIPTION
**Pull request by awinberg-aws**
#### Description of changes

Accessibility review found that having the clear button excluded from the tab order does not meet accessibility standards for keyboard users.

This PR addresses this finding by removing the `excludeFromTabOrder` property on the clear button in the SearchField component

#### Issue #, if available

addresses #5375 

#### Description of how you validated changes
changes validated locally by building and verifying the updated behavior in local instance of the docs site

https://github.com/aws-amplify/amplify-ui/assets/100880084/153fec20-f408-4fa2-b047-a05a8e201f3d
#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [x] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by reesscot — APPROVED** [review]
LGTM

File: packages/react/src/primitives/SearchField/SearchField.tsx [hunk 1]
```diff
@@ -78,7 +78,6 @@ const SearchFieldPrimitive: Primitive<SearchFieldProps, 'input'> = (
       innerEndComponent={
         <FieldClearButton
           ariaLabel={clearButtonLabel}
-          excludeFromTabOrder
```
Comment (calebpollman) [inline packages/react/src/primitives/SearchField/SearchField.tsx:81]: Can we add an a11y unit test asserting behavior that ensures that `excludeFromTabOrder` is not added back in the future?

Comment (awinberg-aws) [reply packages/react/src/primitives/SearchField/SearchField.tsx:81]: Yes! I have a test written for this now but currently blocked from running tests on cloud desktop due to GLIBC_2.28 not being present in AL2 images.

File: packages/react/src/primitives/SearchField/__tests__/SearchField.test.tsx [hunk 2]
```diff
@@ -214,6 +220,23 @@ describe('SearchField component', () => {
       expect(searchField).toHaveFocus();
     });
 
+    it('should be focusable via keyboard', async () => {
```
Comment (calebpollman) [inline packages/react/src/primitives/SearchField/__tests__/SearchField.test.tsx:223]: Ran the test localy for @awinberg-aws

**Review by calebpollman — APPROVED** [review]

File: packages/react/src/primitives/SearchField/SearchField.tsx (hunk 1 above)
Comment (awinberg-aws) [reply packages/react/src/primitives/SearchField/SearchField.tsx:81]: Unit test verified via @calebpollman due to inability to run locally

**Review by reesscot — APPROVED** [review]

**Review by hbuchel — APPROVED** [review]

**Review by timngyn — APPROVED** [review]

**Review by reesscot — APPROVED** [review]

**Pull request merged**